core/peerset/src/lib.rs

```diff
@@ -607,49 +607,55 @@ mod tests {
 		let bootnode2 = PeerId::random();
 		let config = PeersetConfig {
 			in_peers: 0,
-			out_peers: 1,
+			out_peers: 2,
 			bootnodes: vec![bootnode.clone(), bootnode2.clone()],
 			reserved_only: false,
 			reserved_nodes: Vec::new(),
 		};
 
 		let (peerset, handle) = Peerset::from_config(config);
-		handle.report_peer(bootnode2, -1);
+		let peerset = assert_messages(peerset, vec![
+			Message::Connect(bootnode.clone()),
+			Message::Connect(bootnode2.clone()),
+		]);
+
+		handle.report_peer(bootnode2.clone(), -1);
 		handle.report_peer(bootnode.clone(), -1);
 
 		assert_messages(peerset, vec![
-			Message::Connect(bootnode.clone()),
-			Message::Drop(bootnode)
+			Message::Drop(bootnode2),
+			Message::Drop(bootnode),
 		]);
 	}
 
 	#[test]
 	fn test_peerset_report_invulnerable() {
 		let bootnode = PeerId::random();
 		let bootnode2 = PeerId::random();
-		let reserved_node = PeerId::random();
+		let reserved_peer = PeerId::random();
 		let config = PeersetConfig {
 			in_peers: 0,
-			out_peers: 1,
+			out_peers: 2,
 			bootnodes: vec![bootnode.clone(), bootnode2.clone()],
 			reserved_only: false,
-			reserved_nodes: vec![reserved_node.clone()],
+			reserved_nodes: vec![reserved_peer.clone()],
 		};
 
-		let (mut peerset, handle) = Peerset::from_config(config);
-		handle.report_peer(bootnode2, -1);
-		handle.report_peer(reserved_node.clone(), -1);
-
+		let (peerset, handle) = Peerset::from_config(config);
+		let peerset = assert_messages(peerset, vec![
+			Message::Connect(reserved_peer.clone()),
+			Message::Connect(bootnode.clone()),
+		]);
+		
+		handle.report_peer(reserved_peer.clone(), -1);
+		handle.report_peer(bootnode.clone(), -1);
 
-		while !peerset.message_queue.is_empty() {
-			let (message, p) = next_message(peerset).expect("Oh no");
-			assert_ne!(message, Message::Drop(reserved_node.clone()));
-			peerset = p;
-		}
+		// if the reserved_peer is dropped, then the assert_messages will failed
+		assert_messages(peerset, vec![
+			Message::Drop(bootnode)
```

Comment:
Do we need to add one more `Message::Connect(reserved_peer)`?

Reply:
Actually, there is only one message after try to report both reserved_peer and bootnode which is what we expect

Reply:
Oh nvm! `Message::Drop` won't happen at all for reserved_peer and will stay connected